### PR TITLE
Changes to lists of course users and groups

### PR DESCRIPTION
**Pull request by 0xf8f8ff**
- `DynamicTable` and `BootstrapButton` classes now allow custom class names
- table columns in group and user lists are now aligned
- actions in group list are now displayed as buttons, instead of a dropdown menu

image

### CODE REVIEW

File: public/src/components/teacher/CourseGroup.tsx [hunk 1]
```diff
@@ -229,6 +236,60 @@ export class CourseGroup extends React.Component<ICourseGroupProps, ICourseGroup
...
+            case Group.GroupStatus.PENDING:
+                links.push({
+                    name: "Approve",
+                    extra: "primary",
+                    uri: "approve",
+                    description: "list-btn",
+                }, {
+                    name: "Edit",
+                    extra: "primary",
+                    uri: "edit",
```
Comment (meling) [inline public/src/components/teacher/CourseGroup.tsx]: Shouldn't this also have a description?

File: public/src/components/teacher/CourseGroup.tsx [hunk 2]
```diff
@@ -229,6 +236,60 @@ export class CourseGroup extends React.Component<ICourseGroupProps, ICourseGroup
...
+            default:
+                console.log("Got unexpected group status " + group.getStatus() + " when generating links");
+        }
+        return links;
+    }
+
+    private flipEditState() {
+        this.setState({
+            editing: !this.state.editing,
+
```
Comment (meling) [inline public/src/components/teacher/CourseGroup.tsx]: Remove empty line

File: public/src/pages/views/MemberView.tsx [hunk 3]
```diff
@@ -175,7 +175,7 @@ export class MemberView extends React.Component<IUserViewerProps, IUserViewerSta
 
     private async handlePromote(userRel: IUserRelation) {
         if (confirm(
-            `Warning! 
+            `Warning!
```
Comment (meling) [inline public/src/pages/views/MemberView.tsx]: Please change this question to; and remove the `Warning!`:
`Are you sure you want to promote XXX to teacher status?`

File: public/src/components/teacher/CourseGroup.tsx [hunk 4]
```diff
@@ -229,6 +236,60 @@ export class CourseGroup extends React.Component<ICourseGroupProps, ICourseGroup
...
+                    uri: "edit",
+                    description: "list-btn",
+                }, {
+                    name: "Delete",
+                    extra: "danger",
+                    uri: "delete",
+                    description: "list-btn",
+                }) : links.push({
+                    name: "Approved",
+                    extra: "light",
```
Comment (meling) [inline public/src/components/teacher/CourseGroup.tsx]: Shouldn't this have `uri` field?

File: public/src/components/teacher/CourseGroup.tsx (hunk 1 above)
Comment (0xf8f8ff) [reply public/src/components/teacher/CourseGroup.tsx]: None of these needs a description, as we apply the style to the whole button group, so I've removed it for all list buttons.

File: public/src/components/teacher/CourseGroup.tsx (hunk 2 above)
Comment (0xf8f8ff) [reply public/src/components/teacher/CourseGroup.tsx]: Done.

File: public/src/pages/views/MemberView.tsx (hunk 3 above)
Comment (0xf8f8ff) [reply public/src/pages/views/MemberView.tsx]: Fixed.

File: public/src/components/teacher/CourseGroup.tsx (hunk 4 above)
Comment (0xf8f8ff) [reply public/src/components/teacher/CourseGroup.tsx]: No, it's those buttons: 
image

They don't have any uri links.

**Pull request merged**